### PR TITLE
NFDIV-2209 - Updating applicant2 email joint application

### PR DESCRIPTION
**Pull request by danielbe11**
### Change description ###

- Give applicant 1 ability to update applicant 2 email address in joint application.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2209

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/steps/applicant1/their-email-address/content.ts [hunk 1]
```diff
@@ -39,9 +40,10 @@ const en = ({ partner, isDivorce, isJointApplication, hasEnteredSolicitorDetails
       sameEmail: `You have entered your own email address. You need to enter your ${partner}'s email address before continuing.`,
     },
   },
+  continueOrResend: isApplicant2EmailUpdatePossible(userCase) ? 'Resend email' : 'Continue',
 });
 
-const cy: typeof en = ({ partner }) => ({
```
Comment (danielbe11) [inline src/main/steps/applicant1/their-email-address/content.ts:44]: Add back

File: src/main/steps/applicant1/you-cannot-update-their-email/content.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+});
+
+const cy: typeof en = ({ partner }: CommonContent) => ({
+  title: 'You cannot update their email address',
+  alreadyAccessed: `Your ${partner} has already accessed the case online so you cannot update their email address.`,
+  anotherNotification:
+    'You will receive another notification when they have reviewed and confirmed the joint application.',
+  wrongPerson:
+    'If you think the wrong person received the email and accessed the case, please contact the service using the details below.',
+}); //todo
```
Comment (danielbe11) [inline src/main/steps/applicant1/you-cannot-update-their-email/content.ts:20]: Add ticket number

File: src/main/steps/applicant1/email-resent/content.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  reviewApplicationDeadline: `They should review your application and provide some further information by ${userCase.dueDate}.`,
+  yourNotification:
+    'You will receive a notification when they have reviewed. If they do not review then you will be told what you can do to progress the application.',
+  updatePartnersEmailAddress: {
+    part1: 'If you need to, you can ',
+    part2: `update your ${partner}'s email address and resend the email`,
+    part3: '.',
+    link: isApplicant2EmailUpdatePossible(userCase) ? THEIR_EMAIL_ADDRESS : YOU_CANNOT_UPDATE_THEIR_EMAIL,
+  },
+}); //todo
```
Comment (lara-mpt) [inline src/main/steps/applicant1/email-resent/content.ts:32]: if there's no welsh at all probably better to do 
// @TODO translations
const cy: typeof en = en;

File: src/main/steps/applicant1/their-email-address/content.ts [hunk 4]
```diff
@@ -39,9 +40,10 @@ const en = ({ partner, isDivorce, isJointApplication, hasEnteredSolicitorDetails
...
+const cy = ({ userCase, partner }) => ({
```
Comment (lara-mpt) [inline src/main/steps/applicant1/their-email-address/content.ts:46]: worth keeping in 
const cy: typeof en = ({ partner }) => ({

File: src/main/steps/applicant1/their-email-address/post.test.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { mockRequest } from '../../../../test/unit/utils/mockRequest';
+import { mockResponse } from '../../../../test/unit/utils/mockResponse';
+import { Checkbox } from '../../../app/case/case';
+import { CITIZEN_RESEND_INVITE, CITIZEN_UPDATE } from '../../../app/case/definition';
+import { FormContent } from '../../../app/form/Form';
+import * as contentUtils from '../../common/content.utils';
```
Comment (lara-mpt) [inline src/main/steps/applicant1/their-email-address/post.test.ts:6]: better to import individually

File: src/main/steps/applicant1/their-email-address/post.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import { AnyObject, PostController } from '../../../app/controller/PostController';
+import { isApplicant2EmailUpdatePossible } from '../../common/content.utils';
+
+@autobind
+export default class TheirEmailAddressPostController extends PostController<AnyObject> {
+  protected getEventName(req: AppRequest<AnyObject>): string {
+    if (isApplicant2EmailUpdatePossible(req.session.userCase)) {
+      return CITIZEN_RESEND_INVITE;
+    } else {
+      return CITIZEN_UPDATE;
```
Comment (lara-mpt) [inline src/main/steps/applicant1/their-email-address/post.ts:14]: could return super.getEventName() here

File: src/main/steps/applicant1/you-cannot-update-their-email/content.ts (hunk 2 above)
Comment (lara-mpt) [reply src/main/steps/applicant1/you-cannot-update-their-email/content.ts:20]: if there's no welsh at all probably better to do
// @todo translations
const cy: typeof en = en;

**Review by lara-mpt — APPROVED** [review]

File: src/main/steps/applicant1/their-email-address/content.ts (hunk 4 above)
Comment (danielbe11) [reply src/main/steps/applicant1/their-email-address/content.ts]: done

File: src/main/steps/applicant1/you-cannot-update-their-email/content.ts (hunk 2 above)
Comment (danielbe11) [reply src/main/steps/applicant1/you-cannot-update-their-email/content.ts]: done

File: src/main/steps/applicant1/their-email-address/content.ts (hunk 1 above)
Comment (danielbe11) [reply src/main/steps/applicant1/their-email-address/content.ts:44]: done

**Pull request closed without merging**